lib/utils.js

```diff
@@ -0,0 +1,315 @@
+               'name': exits[j],
+            };
+            if (vars_equiv(node, temp_exit)){
+               return true;
+            }
+         }
+   }
+
+   return false;
+}
+
+function has_exit(node, function_names){
+   /*
+      Checks function bodies for an exit call.
+
+      @param node: an esprima node with `body` property or an array of esprima 
+         nodes
+      @param function_names: an array of names of functions that always exit
+
+      @return: a bool of whether `node` always exits
+   */
+   var actual_body = [];
+   if (node && node.body){
+      actual_body = node.body;
+      if (actual_body.body){
+         actual_body = actual_body.body;
+      }
+   }
+   for (var i = 0; i < actual_body.length; i++) {
+      if (is_exit(actual_body[i], function_names)){
+         return true;
+      }
+   }
+
+   return false;
+}
+
+function nodes_equal(node1, node2){
+   /* 
+      Checks whether two nodes are equal by comparing the range of code each node covers.
+
+      @param node1: an esprima node with range property
+      @param node2: an esprima node with range property
+
+      @return: a bool of whether the nodes are equal
+   */
+
+   // range is an array with two elements
+   if (node1.range){
+      if (node1.range[0] === node2.range[0]){
+         if (node1.range[1] === node2.range[1]){
+            return true;
+         }
+      }
+   }
+   return false;
+}
+
+function get_original_node(node, filetext){
+   /*
+      Quick helper for extracting code as string.
+
+      @param node: an esprima node with range property
+      @param filetext: the string of the complete code from which `node` was 
+         parsed
+
+      @return: a string of the original code
+   */
+   return filetext.substring(node.range[0], node.range[1]);
+}
+
+function check_functions(node, exits){
+   /*
+      Extracts all functions which always exit.
+
+      Used to handle the "wrapped exit call" case.
+
+      @param node: an esprima node with `body` property that is an array
+      @param exits: any functions already identified as exiting
+
+      @return: an array of those functions which always exit
+   */
+   var exiting = [];
+   if (node && Array.isArray(node.body)){
+      for (var i = 0; i < node.body.length; i++) {
+         var child = node.body[i];
+         if (child.type === 'FunctionDeclaration'){
+            var actual_body = child.body.body;
+            for (var j = 0; j < actual_body.length; j++) {
+               if (is_exit(actual_body[j], exits)){
+                  exiting.push(child.id.name);
+               }
+            }
+         }
+      }
+   }
+
+   return exiting;
+}
+
+function vars_equiv(var1, var2){
+   /*
+      Checks whether two esprima variable nodes represent the same variable.
+
+      @param var1: an esprima node for a variable
+      @param var2: an esprima node for a variable
+
+      @return: boolean of whether `var1` and `var2` represent the same variable
+   */
+   if (var1 && var2){
```

Comment:
Again, if you find yourself `returning true`, then maybe you can just return the entire boolean expression